### PR TITLE
Tabs to spaces v3.0

### PR DESCRIPTION
**Pull request by amrondonp**
Change tabs to spaces before the change and only checking the code that is being inserted in that change. So no considerable efficiency issues added

### CODE REVIEW

File: inginious/frontend/common/static/js/common.js [hunk 1]
```diff
@@ -93,6 +93,14 @@ function registerCodeEditor(textarea, lang, lines)
         codeMirrorInstance.replaceSelection(spaces);
     }
 
+    var fixTabs = function(codeMirrorInstance, changeObj){
+        var indentUnit = codeMirrorInstance.getOption("indentUnit");
+        var spaces = Array(indentUnit + 1).join(" ");
+        changeObj.text = $.map(changeObj.text, function(line) {
+            return line.replace(/\t/g, spaces);
```
Comment (0xDCA) [inline inginious/frontend/common/static/js/common.js:100]: nit: Do we actually need to use a regular expression here? Why not a string? It should be slightly more efficient

**Review by 0xDCA — COMMENTED** [review]
A minor comment

File: inginious/frontend/common/static/js/common.js (hunk 1 above)
Comment (amrondonp) [reply inginious/frontend/common/static/js/common.js:100]: For future obvservers: 
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/replace

"\t" only would replace the first occurrence

Comment (0xDCA) [reply inginious/frontend/common/static/js/common.js:100]: Ignore that, it will only replace the first match if the first argument is a String

**Comment by 0xDCA** [comment]
LGTM

**Pull request merged**